app/controllers/friend_requests_controller.rb

```diff
@@ -0,0 +1,37 @@
+
+class FriendRequestsController < ApplicationController
+  before_action :validate_creation, only: :create
+
+  def index
+    @requests = current_user.friend_requests
+  end
+
+  def create
+    friend_request = current_user.sent_requests.build(friend_request_params)
+    if friend_request.save
+      flash[:succes] = 'Friend request sent'
+    else
+      flash[:alert] = 'Unable to send request'
+    end
+    redirect_back fallback_location: users_path
+  end
+
+  def destroy
+    request = FriendRequest.find(params[:id])
+    request.destroy
+    redirect_back fallback_location: friend_requests_path
+  end
+
+  private
+
+  def friend_request_params
+    params.require(:friend_request).permit(:receiver_id)
+  end
+
+  def validate_creation
```

Comment:
Hi guys. I mentioned something related to this in my previous review. I think we can transfer this validation in `friend_request.rb`. We can simply add a `before_validations` callback in there that will check whether a friendship already exists between the `sender` and `receiver`. If there is an existing friendship then we can add an error. This error then will make possible the ff:

```ruby
# supose there is already an existing friendship between kahwi and serge

request = kahwi.sent_requests.build(receiver: serge)
request.valid? # returns false
request.errors.full_messages # should show "Already friends, you cannot send another request to this user"
```
We can then use such error to be displayed as a flash message (like what you're already doing but instead of hard coding "Unable to send request" you can just  use the error message).

```ruby
def create
 request = current_user.sent_request.build(friend_request_params)
 if request.save?
   # do something nice
 else
   flash[:error] = request.errors.full_messages[0]
 end
end
```